src/internal/Get-MsGraphResults.ps1

```diff
@@ -292,20 +338,46 @@ function Get-MsGraphResults {
                 $MsGraphSession = Confirm-ModuleAuthentication -MsGraphSession -ErrorAction Stop
                 if (!$NoAppInsights) { $Stopwatch = [System.Diagnostics.Stopwatch]::StartNew() }
                 try {
-                    # [hashtable] $results = Invoke-MgGraphRequest -Method $Request.method -Uri $uriEndpoint.AbsoluteUri -Headers $Request.headers
-                    $results = Invoke-RestMethod -WebSession $MsGraphSession -UseBasicParsing @paramInvokeRestMethod -ErrorAction Stop
-                    if ($IncapsulateReferenceListInParentObject -and $Request.url -match '.*/(.+)/(.+)/((?:transitive)?members|owners)') {
-                        [PSCustomObject]@{
-                            id            = $Matches[2]
-                            '@odata.type' = '#{0}' -f (Get-MsGraphEntityType $GraphBaseUri.AbsoluteUri -EntityName $Matches[1])
-                            $Matches[3]   = Complete-MsGraphResult $results -DisablePaging:$DisablePaging -KeepODataContext:$KeepODataContext -AddODataType:$AddODataType -GroupOutputByRequest -Request $Request -GraphBaseUri $GraphBaseUri
+                    for($Retries = 0; $Retries -lt $MaxRetries; $Retries++) {
+                        try {
+                            $results = Invoke-RestMethod -WebSession $MsGraphSession -UseBasicParsing @paramInvokeRestMethod -ErrorAction Stop
+                            # break the loop if no error was raised
+                            break
                         }
+                        catch {
+                            ## error while invoking graph
+                            if ($Retries -eq $MaxRetries-1) {
+                                # catch error if it was the last try
+                                Catch-MsGraphError $_ 
+                            }
+                            ## check if throttling happened
+                            if ($_.Exception.PSobject.Properties.Name.Contains("Response") -and $_.Exception.Response.StatusCode.value__ -eq 429) {
+                                # Get the retry after header
+                                try {
+                                    $RetryAfter = [double]($_.Exception.Response.Headers.GetValues('Retry-After')[0])
```

Comment:
Has this been tested on both PS7 and Windows PS 5.1? I know the exception object is different so just wanting to double check.

Reply:
i tested on PS7 only will validate on PS5.1

Reply:
Brought some updates for issues seen on PS5.1 (mostly arround app credential & certs)
Didn't have specific throttling. in this code if the Exception doesn't have a "Response" it should fallback to the default timings.